File: driver.hs
import qualified System.Directory as D
import qualified System.IO.Unsafe as SIU

outputDir = "data"
rawExt = ".xml"


mapMI :: (a -> IO b) -> [a] -> IO [b]
mapMI _ [] 	= return []
mapMI f (x:xs) 	= do y <- SIU.unsafeInterleaveIO (f x) ; ys <- SIU.unsafeInterleaveIO (mapMI f xs) ; return (y:ys)

processFileContent = length

getBugs = do
        files <- D.getDirectoryContents outputDir
        let xmlFiles = map (\n -> outputDir ++ "/" ++ n) $
                       filter (\n -> (reverse . take 4 . reverse) n == rawExt) files
        mapMI (\f -> do {str <- readFile f; return (processFileContent str); }) (xmlFiles)


main = do
	x <- getBugs
	putStrLn (show x)

File: driver.pl
#!/usr/bin/perl

use strict;
use warnings;

my$outputDir = "data";
my$rawExt = ".xml";

sub processFileContent {
	my($str)=@_;
	return (length $str);
}

sub getBugs{
	my@files = glob("$outputDir/*.xml");
	my@result;
	for my $file (@files) {
		open FILE, $file;
		undef local $/;
		my$slurp=<FILE>;
		push @result, (processFileContent($slurp));
	}
	return \@result;
}

my$res=getBugs();
print "@$res";

